src/resolver.ts

```diff
@@ -2637,9 +2637,18 @@ export class Resolver extends DiagnosticEmitter {
     ) {
       // (x) => ret, infer return type accordingt to `ret`
       const expr = (<ExpressionStatement>body).expression;
-      const type = this.resolveExpression(expr, ctxFlow, ctxType, reportMode);
+      let signatureReference = assert(functionType.getSignature());
+      // create a temp flow to resolve expression
+      let tempFlow = Flow.createParent(ctxFlow.actualFunction);
+      let parameters = signature.parameters;
+      assert(signatureReference.parameterTypes.length == parameters.length);
```

Comment:
Can this assert potentially fail? Iirc, function expressions can omit arguments. `Array#forEach` for example is defined as `forEach(fn: (value: T, index: i32, array: Array<T>) => void): void`, and it is sufficient to use it as `.forEach(value => ...)`.

Reply:
in `resolveFunctionType`, `functionType.getSignature()` is generated by `parameters`. So the length must be same.
But it is guarantee by the `resolveFunctionType`. So I add a assert here to for quick find the bug when `resolveFunctionType` change its behavior.

Reply:
Can you add a comment? :)